projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -410,15 +410,23 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
             return;
         }
         // this.state.startIndex = index;
+        const containerSize = parseInt(this.igxForContainerSize, 10);
+        const isPrevItem = index < this.state.startIndex;
         if (this.igxForScrollOrientation === 'horizontal') {
-            this.hScroll.scrollLeft = this.sizesCache[index];
+            this.hScroll.scrollLeft = isPrevItem ?
+            this.sizesCache[index] :
+            this.sizesCache[index] - containerSize + this.igxForOf[index].width;
         } else {
-            const containerSize = parseInt(this.igxForContainerSize, 10);
             const maxVirtScrollTop = this._virtHeight - containerSize;
-            let nextScrollTop = this.sizesCache[index];
+            let nextScrollTop = isPrevItem ?
+            this.sizesCache[index] :
+            this.sizesCache[index] - containerSize + this.heightCache[index];
```

Comment:
Again a case where the expression evaluates less than 0? I see there is a check below but in that case the igxFor won't scroll the whole item into view.